sqladmin/fields.py

```diff
@@ -425,3 +427,92 @@ def pre_validate(self, form: Form) -> None:
                 identity = inspect(v).identity
                 if identity and str(identity[0]) not in pk_list:  # pragma: no cover
                     raise ValidationError(self.gettext("Not a valid choice"))
+
+
+class AjaxSelectField(SelectFieldBase):
+    """Ajax Model Select Field"""
+
+    widget = sqladmin_widgets.AjaxSelect2Widget()
+
+    separator = ","
+
+    def __init__(
+        self,
+        loader,
+        label=None,
+        validators=None,
+        allow_blank=False,
+        blank_text="",
+        object_list=None,
+        **kwargs,
+    ):
+        super().__init__(label, validators, **kwargs)
+        self.loader = loader
+
+        self.allow_blank = allow_blank
+        self.blank_text = blank_text
+
+    @property
+    def data(self) -> Any:
+        if self._formdata:
+            self.data = self._formdata
```

Comment:
The most important change needed is that here we're setting the PK of that object inside in the form, which as part of the design we expect to set the SQLAlchemy object, not the PK only.

the challenge is that it's not possible to query the async db in the sync WTForm method. 
The only way I could think of is to handle this when inserting the object.

inside the insert we can check if this attribute name is a relationship and if it is, query based on that PK to get the SQLAlchemy object with it. It's a bit hackish but I'll look for other approaches.

Reply:
It would be great if @dwreeves could take a look.

Reply:
Sorry, started a new job this week. I will take a look this weekend!